### PR TITLE
[quant][graphmode][fx] Add support for fp16 bmm pattern

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52865 [quant][graphmode][fx][fp16] Add fp16 support for silu
* #52864 [quant][graphmode][fx][fp16] Add fp16 support for tanh
* #52863 [quant][graphmode][fx][fp16] Add fp16 support for sigmoid
* #52862 [quant][graphmode][fx][fp16] Add fp16 support for layer_norm
* #52811 [quant][graphmode][fx][fp16] Add fp16 support for torch.sum
* #52810 [quant][graphmode][fx][fp16] Add fp16 support for div
* #52809 [quant][graphmode][fx][fp16] Add fp16 support for sub
* **#52808 [quant][graphmode][fx] Add support for fp16 bmm pattern**

Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

### CODE REVIEW

File: test/quantization/test_quantize_fx.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ class BinaryOpNonQuantizedInput(torch.nn.Module):
     def __init__(self, binary_op, ibinary_op, is_inplace, is_scalar):
         super().__init__()
         self.is_scalar = is_scalar
-        self.op = ibinary_op if is_inplace else binary_op
+        self.op = ibinary_op if ibinary_op and is_inplace else binary_op
```
Comment (vkuzo) [inline test/quantization/test_quantize_fx.py:107]: optional: what's the difference between binary_op or ibinary_op?  Would be great if there was a comment

File: torch/quantization/fx/quantization_patterns.py [hunk 2]
```diff
@@ -109,9 +110,12 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
             torch.mul: torch.ops.quantized.mul_relu,
         }
         # corresponding quantized op
-        self.qop = qbin_relu_op_mapping[self.bop] \
-            if self.relu_node is not None \
-            else qbin_op_mapping[self.bop]  # type: ignore
+        if self.bop in qbin_op_mapping:
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:113]: optional nit (probably for an earlier PR): `bop` is hard to think about, maybe `binary_op`?  same for `qop`

**Review by vkuzo — APPROVED** [review]

File: test/quantization/test_quantize_fx.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:109]: oh that means inplace binary op

File: torch/quantization/fx/quantization_patterns.py (hunk 2 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:114]: sure
